vmdb/app/models/ems_amazon.rb

```diff
@@ -49,13 +49,18 @@ def verify_credentials(auth_type=nil, options={})
     begin
       # EC2 does Lazy Connections, so call a cheap function
       with_provider_connection(options.merge(:auth_type => auth_type)) { |ec2| ec2.regions.map(&:name) }
-    rescue AWS::EC2::Errors::SignatureDoesNotMatch
+    rescue AWS::EC2::Errors::SignatureDoesNotMatch => err
+      $log.error("MIQ(#{self.class.name}.verify_credentials) Error Class=#{err.class.name}, Message=#{err.message}")
       raise MiqException::MiqHostError, "SignatureMismatch - check your AWS Secret Access Key and signing method"
-    rescue AWS::EC2::Errors::AuthFailure
+    rescue AWS::EC2::Errors::AuthFailure => err
+      $log.error("MIQ(#{self.class.name}.verify_credentials) Error Class=#{err.class.name}, Message=#{err.message}")
       raise MiqException::MiqHostError, "Login failed due to a bad username or password."
+    rescue AWS::Errors::MissingCredentialsError => err
+      $log.error("MIQ(#{self.class.name}.verify_credentials) Error Class=#{err.class.name}, Message=#{err.message}")
+      raise MiqException::MiqHostError, "Missing credentials"
     rescue Exception => err
       $log.error("MIQ(#{self.class.name}.verify_credentials) Error Class=#{err.class.name}, Message=#{err.message}")
-      raise MiqException::MiqHostError, "Unexpected response returned from system, see log for details"
+      raise MiqException::MiqHostError, "Unexpected response returned from system: #{err.message}"
     end
```

Comment:
@himdel Ultimately this looks similar to the logic that `translate_exception` does for the database adapters for translating one exception into another with a custom message:

http://apidock.com/rails/v4.1.8/ActiveRecord/ConnectionAdapters/PostgreSQLAdapter/translate_exception

We don't need to implement that now, but it would be nice to have a test that would make that easy to do.

For example, we can start with something like this:

```ruby
ems.stub(:with_provider_connection).and_raise(AWS::EC2::Errors::SignatureDoesNotMatch, "blah").
...
$log.should_receive(:warn).with(/blah/)`
...
lambda { ems.verify_credentials(args).should raise_error(MiqException::MiqHostError, "SignatureMismatch - check your AWS Secret Access Key and signing method)
```

But ultimately, tests and code like this are brittle and it would be better to instead do something like this:

```ruby
     begin
       # EC2 does Lazy Connections, so call a cheap function
       with_provider_connection(options.merge(:auth_type => auth_type)) { |ec2| ec2.regions.map(&:name) }
    rescue => err
      miq_exception = translate_exception(err)
      raise unless miq_exception # untranslated exceptions are re-raised
      $log.error("MIQ(#{self.class.name}.verify_credentials) Error Class=#{err.class.name}, Message=#{err.message}")
      raise miq_exception
    end
```

We can then easily test `translate_exception` as it's now a query method, provide an exception, it returns another one.